### DISCUSSION TITLE
Do we have PixiJS v7 feautres list?

### DISCUSSION THREAD

**Discussion by goldenratio** [pixijs/pixijs#8665] [Q&A] [answered]
just curious.

**Comment by bigtimebuddy** [comment, marked as answer]
In general the "features" are pretty light in v7, it's largely a modernization update. There are things in PixiJS that hadn't been updated in a long time. Here's a quick round-up of the big changes that are likely to impact users the most:

## Dropping Internet Explorer

Microsoft officially ended support for IE, so we decided to follow. It simplified a lot of our modernizations since IE was an outliner from Safari/Chrome/Firefox/Edge and mobile browser.

## Remove Polyfills 

We removed the bundled polyfills such as `requestAnimationFrame` and `Promise`. These things are widely available in browsers now. If projects require them, developers should include the polyfills they need for backward-compatibility.

## Output ES2020 (modules) and ES2017 (browser)

PixiJS historically only output in ES5 (no classes!). This new output standard allows us to at least use ES2017 features that previously we couldn't use, like `String.prototype.startsWith`, `Array.prototype.contains`, etc. No only does it make the code more readable, but the output looks much nicer as well. For modules, we are outputting ES2020, which has things like nullish coalescing. If your project needs to have backward support, you can use Babel to easily transpile these things or polyfill.

## Replaces InteractionManager with EventSystem

InteractionManager was getting extremely complex and difficult to maintained. Very few people really understood it. We decided to move to FederatedEvents, which is more concise, better aligned with the DOM and supports things like bubbling that previously the InteractionManager didn't have. The great news, is you shouldn't have to change code, as it is largely a drop-in replacement. We added `addEventListener` and `removeEventListener` APIs to DisplayObject which have the same DOM signature. These can be used instead of `on` and `off`.

## Replaces Loader with Assets

Similarly, we've been wanting to remove the Loader because of it's legacy approach (e.g., XMLHttpRequest). This was forked from [resource-loader](https://github.com/englercj/resource-loader) the awesome package from @englercj that has been with PixiJS for a long time. The original design inspiration for Loader was driven largely by Flash/AS3, which now seem dated. There were a few things we wanted out of a new iteration: static loading, loading with Workers, background loading, Promise-based, fewer layers of caching. Here's a quick example of how this will change:

```js
import { Loader, Sprite } from 'pixi.js';

const loader = new Loader();
loader.add('background', 'path/to/assets/background.jpg');
loader.load((loader, resources) => {
  const image = Sprite.from(resources.background.texture);
});
```

```js
import { Assets, Sprite } from 'pixi.js';

const texture = await Assets.load('path/to/assets/background.jpg');
const image = Sprite.from(texture);
```

## Abandon the use of peerDependencies

PixiJS heavily uses peerDependencies in the **package.json** of each package. This design choice has plagued Pixi with all sorts of issues. It's a breaking change to remove, so now was a good time. We have decided to completely remove peerDependencies, instead opting for _nothing_. This should make installing and upgrading `pixi.js` much easier. We are working on some updating tooling for composing a custom version with packages.

**Reply by mcdenhoed** [reply]
Just wanted to say I love these changes and I love what you all have done with Pixi! I've been using Pixi for almost 4 years now and the steady, well telegraphed, modernization effort has been very much appreciated. Can't wait for this to land!

**Comment by domis86** [comment]
@bigtimebuddy good info. This should be added to https://github.com/pixijs/pixijs/wiki maybe?

**Reply by bigtimebuddy** [reply]
I'm working on a migration guide for the wiki, which should have all this info and more.

**Reply by bigtimebuddy** [reply]
Here's the start, feedback welcome:
https://github.com/pixijs/pixijs/wiki/v7-Migration-Guide
